client/src/modules/blockEditor/actions.ts

```diff
@@ -114,11 +177,24 @@ export const removeExportOfSelection = () => {
         return;
       }
 
-      const change = block.value
-        .change()
-        .unwrapInlineByKey(matchingNodes[0].key, {
-          data: { pointerId: hoveredItem.id }
-        });
+      const nodeToRemove = matchingNodes[0];
+
+      let change;
+      change = block.value.change();
+
+      // remove pointerExport inline
+      const ancestorNodes = block.value.document.getAncestors(nodeToRemove.key);
+      const isNested = ancestorNodes.find(ancestor => ancestor.type === "pointerExport");
+
+      if (isNested) {
+        change
+          .unwrapInlineByKey(nodeToRemove.key,  { data: { pointerId: hoveredItem.id } })
+          .removeNodeByKey(nodeToRemove.key);
+      } else {
+        change.unwrapInlineByKey(nodeToRemove.key,  { data: { pointerId: hoveredItem.id } });
+      }
+
+      normalizeAfterRemoval(change);
```

Comment:
I feel like the above functions have become quite long. I'm more uncomfortable with messiness in the actions than I am in other areas, because these can be a pain to debug. 

Perhaps there could be a pattern for mutations to change. I could imagine code like,
```js
changeMutations = [correctWrapping, normalizeAfterRemoval]
applyChanges(value, changeMutations)
```